### PR TITLE
Update API docs on saturation, rewards, and relative stake

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#2155 

# Overview

- [x] Tweaked API docs

# Comments

- Not sure if we should rush this to be in the release or not

### CODE REVIEW

File: specifications/api/swagger.yaml [hunk 1]
```diff
@@ -723,12 +723,26 @@ x-stakePoolMargin: &stakePoolMargin
 
       May be omitted if the wallet hasn't found the pool's registration cerificate yet.
 
+x-jormungandrStakePoolSaturation: &jormungandrStakePoolSaturation
```
Comment (Anviking) [inline specifications/api/swagger.yaml:726]: The old definition. I made a new extended Shelley one.

File: specifications/api/swagger.yaml [hunk 2]
```diff
@@ -723,12 +723,26 @@ x-stakePoolMargin: &stakePoolMargin
...
+  example: 0.74
+
 x-stakePoolSaturation: &stakePoolSaturation
   type: number
   minimum: 0
   description: |
     Saturation-level of the pool based on the desired number of pools aimed by the network.
     A value above `1` indicates that the pool is saturated.
+
+    The `non_myopic_member_rewards` take oversaturation into account, as specified by the [specs](https://hydra.iohk.io/job/Cardano/cardano-ledger-specs/delegationDesignSpec/latest/download-by-type/doc-pdf/delegation_design_spec).
```
Comment (Anviking) [inline specifications/api/swagger.yaml:741]: There was _a lot of_ confusion about this, so let's state it clearly.

File: specifications/api/swagger.yaml [hunk 3]
```diff
@@ -760,11 +774,15 @@ x-stakePoolMetrics: &stakePoolMetrics
...
+      description: |
+        The live pool stake relative to the *total* stake.
+
+        For more details, see the section "Relative Stake: Active vs Total" in
+        [Design Specification for Delegation and Incentives in Cardano](https://hydra.iohk.io/job/Cardano/cardano-ledger-specs/delegationDesignSpec/latest/download-by-type/doc-pdf/delegation_design_spec).
     saturation: *stakePoolSaturation
     produced_blocks:
       <<: *numberOfBlocks
-      description: Number of blocks produced by a given stake pool
+      description: Number of blocks produced by a given stake pool in its lifetime.
```
Comment (Anviking) [inline specifications/api/swagger.yaml:785]: > in its lifetime.

For added clarity. But this is not valid for jormungandr. Need to differentiate.

File: specifications/api/swagger.yaml [hunk 4]
```diff
@@ -760,11 +774,15 @@ x-stakePoolMetrics: &stakePoolMetrics
     non_myopic_member_rewards: *nonMyopicMemberRewards
     relative_stake:
       <<: *percentage
-      description: The pools stake compared to the stake of all pools.
+      description: |
+        The live pool stake relative to the *total* stake.
```
Comment (Anviking) [inline specifications/api/swagger.yaml:778]: > total

With cardano-node 1.19 and older, this used to be _active_.

File: specifications/api/swagger.yaml [hunk 5]
```diff
@@ -723,12 +723,26 @@ x-stakePoolMargin: &stakePoolMargin
...
   type: number
   minimum: 0
   description: |
     Saturation-level of the pool based on the desired number of pools aimed by the network.
     A value above `1` indicates that the pool is saturated.
+
+    The `non_myopic_member_rewards` take oversaturation into account, as specified by the [specs](https://hydra.iohk.io/job/Cardano/cardano-ledger-specs/delegationDesignSpec/latest/download-by-type/doc-pdf/delegation_design_spec).
+
+    The saturation is based on the live `relative_stake`. The saturation at the end of epoch e,
+    will affect the rewards paid out at the end of epoch e+3.
```
Comment (KtorZ) [inline specifications/api/swagger.yaml:744]: :+1:

**Review by KtorZ — APPROVED** [review]
since we want to release on 1.20.0, I'd like to have that merged unless you think there are good reasons to hold on? 

I supposed you checked some of these claims with Jared & al ?

File: specifications/api/swagger.yaml (hunk 2 above)
Comment (Anviking) [reply specifications/api/swagger.yaml:741]: _This is checked._

File: specifications/api/swagger.yaml (hunk 5 above)
Comment (Anviking) [reply specifications/api/swagger.yaml:744]: I've inferred these, but _not 100% confirmed_
- [ ] The saturation is based on the live `relative_stake`
- [ ] will affect the rewards paid out at the end of epoch e+3

File: specifications/api/swagger.yaml (hunk 4 above)
Comment (Anviking) [reply specifications/api/swagger.yaml:778]: _main point of the upstream changes, should be true_

Comment (Anviking) [reply specifications/api/swagger.yaml:778]: > live pool stake

I read this, but _not 100% confirmed_

File: specifications/api/swagger.yaml (hunk 5 above)
Comment (Anviking) [reply specifications/api/swagger.yaml:744]: > will affect the rewards paid out at the end of epoch e+3

Well, based on:

Skärmavbild 2020-09-22 kl  13 58 40

> The saturation is based on the live relative_stake

Looks pretty clear from a comment in #1854 : 

> This is not based on any snapshot, but uses the current ledger state.

And I noticed differences in ranking when trying out live.

File: specifications/api/swagger.yaml (hunk 3 above)
Comment (Anviking) [reply specifications/api/swagger.yaml:785]: Actually, it is already differentiated. This change is for the shelley one only.

**Comment by Anviking** [comment]
bors r+

> I supposed you checked some of these claims with Jared & al ?

I double-checked with Philipp now.

**Pull request merged**